splash/auth/oauth_resources.py

```diff
@@ -0,0 +1,61 @@
+import jsonschema
+import json
+import os
+from flask import current_app
+from flask import request
+from flask_restful import Resource
+from google.oauth2 import id_token
+from google.auth.transport import requests
+
+
+#TODO: Create error handling for if the user is not found in the mongo database
+#TODO: integrate this with mongo 
+class OauthVerificationError(ValueError):
+    pass
+
+class UserNotFoundError(Exception): #TODO: make the base class more specific
+    pass
+
+
+class OAuthResource(Resource):
+        def __init__(self):
+            self.CLIENT_ID = os.environ.get("GOOGLE_CLIENT_ID") + ".apps.googleusercontent.com" #TODO: Temporary solution, we will need to have a local config file
+            AUTH_SCHEMA = open_schema()
+            self.validator = jsonschema.Draft7Validator(AUTH_SCHEMA)
+        def post(self):
+            try: 
+                payload = request.get_json(force=True)
+                self.validator.validate(payload)
+                token = payload['token']
+                # Specify the CLIENT_ID of the app that accesses the backend:
+                idinfo = id_token.verify_oauth2_token(token, requests.Request(), self.CLIENT_ID)
+
+                # Or, if multiple clients access the backend server:
+                # idinfo = id_token.verify_oauth2_token(token, requests.Request())
+                # if idinfo['aud'] not in [CLIENT_ID_1, CLIENT_ID_2, CLIENT_ID_3]:
+                #     raise ValueError('Could not verify audience.')
+                
+                print(idinfo['iss'] not in ['accounts.google.com', 'https://accounts.google.com'])
```

Comment:
print statements really should never be checked into git unless they are needed to tell the user that there's an error with logging.

I'd like you to at the top of this file obtain a logger 
`
logger = logging.getLogger('splash-server')`

and then log to it for this....I dunno, info level log?

Reply:
oops...I hadn't even looked at what that does. True or false. We can delete that however, I would like to maybe a .info that says that a person was successfully logged in from google, giving their email address at least.

Reply:
Yeah this should be deleted. I only meant it as a debugging statement. oops